server.ts

```diff
@@ -26,11 +26,46 @@ app.use(cors()) //add CORS support to each following route handler
 const client = new Client(dbConfig);
 client.connect();
 
-app.get("/", async (req, res) => {
-  const dbres = await client.query('select * from categories');
-  res.json(dbres.rows);
+app.get("/dogs", async (req, res) => {
+  const dbres = await client.query('select breed from dogs');
+  res.set('Access-Control-Allow-Origin', '*')
+  res.status(200).json(dbres.rows);
 });
 
+app.get("/topten", async (req, res) => {
+  const dbres = await client.query(`SELECT * FROM dogs ORDER BY votes DESC, breed ASC LIMIT 10`)
+  res.set('Access-Control-Allow-Origin', '*')
+  res.status(200).json(dbres.rows)
+})
+
+app.post("/", async (req, res) => {
+  try{
+  if(req.body.breedNames.length>0){
+  await client.query('insert into dogs (votes, breed) values ($1,$2) on conflict (breed) do nothing', ['0', req.body.breedNames[0]])
+  await client.query('insert into dogs (votes, breed) values ($1,$2) on conflict (breed) do nothing', ['0', req.body.breedNames[1]])
+  res.set('Access-Control-Allow-Origin', '*')
+  res.status(200).send({success: true, data: req.body.breedNames})
+  }
+  else{
+    res.status(100).send("connection with server secured")
```

Comment:
The message sent here `connection with server secured` isn't descriptive enough to encapsulate what you wanted to do - maybe a comment explaining would've helped.